src/utils/codec.ts

```
@@ -1141,6 +1141,28 @@ const ENVELOPE_HANDLERS: ReadonlyMap<string, EnvelopeHandler> = new Map([
+  return proto === Object.prototype || proto === null;
+}
+
+function isPlainArray(value: unknown): value is unknown[] {
+  return Array.isArray(value) && Object.getPrototypeOf(value) === Array.prototype;
+}
+
+function decodePath(base: string, key: string | number): string {
+  if (typeof key === 'number') {
+    return `${base}[${key}]`;
+  }
+  return /^[A-Za-z_$][\w$]*$/.test(key) ? `${base}.${key}` : `${base}[${JSON.stringify(key)}]`;
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)